addons/portal/wizard/portal_wizard.py

```diff
@@ -120,7 +148,8 @@ def action_grant_access(self):
         An invitation email will be sent to the partner.
         """
         self.ensure_one()
-        self._assert_user_email_uniqueness()
+        if self.email_validity_state != 'valid':
+            return self.wizard_id._action_open_modal()
```

Comment:
I think we could still raise the errors from before here depending on the email_validity_state value.

Typically, if you try to bypass the interface and call this method directly in the code, you should get an error raised if the email is invalid / already user.

Same thing for "action_revoke_access".

Reply:
@awa-odoo It makes sense indeed, but does go against the initial purpose of the commit. If I raise errors it will still create an error modal over the wizard modal. However, this will only happen if I edit the email then immediately try to grant access. If i leave edit mode of that email, The view will be updated and buttons will disappear. Any thoughts? Cheers

Reply:
Hmm, I see what you mean.

There is a property called "onchange_on_keydown" or something like that that we added on Char fields.
Maybe it would work here?
I'm not sure, maybe it will be annoying because the buttons will disappear as soon as you start typing then.
Probably to try and check how it feels.

I agree that it goes a bit against the initial purpose but the goal is to have a nicer UI *while still keeping robust code* so I would keep those raised errors on the methods in any case.

Reply:
@awa-odoo I tried with the option onchange_on_keydown but It was very slow to notice change - It feels like there is a delay before doing the onchange, maybe 2 - 3 seconds -, and the parent 'has_invalid_emails' is not updated, probably because it is out of scope in the xml. Therefore, I don't know if this is ideal. Should we stay on current way? I could also make the change and make the email status column always visible, but the delay makes it a bit weird. User will probably use buttons directly after entering the email, so they will not have the time to see the onchange removing buttons and will instead trigger an Error. What do you think? Cheers